packages/liveblocks-react/src/factory.tsx

```diff
@@ -346,6 +363,15 @@ export function createRoomContext<
 
     const _client = useClient();
 
+    // NOTE: Boxing deliberately avoids this value from triggering effects
+    // through the exhaustive-deps array.
+    const box = useBox({
+      initialPresence,
+      initialStorage,
+      defaultPresence, // Will get removed in 0.18
+      defaultStorageRoot, // Will get removed in 0.18
+    });
```

Comment:
Here is an interesting/novel construct. The `exhaustive-deps` rule was complaining about the fact that these variables were relied on in the `useEffect()` callback (correct), and that, as such, they should be included in the dependency array.

This makes a lot of sense, but it would have an adverse effect of leaving and reentering the room every time the initial values passed to the `RoomProvider` change. In practice, that will mean every time, because we construct/recommend examples to be structured like this:

```tsx
<RoomProvider
  initialPresence={{ x: 0, y: 0 }}
  initialStorage={() => ({ items: ... })}>
```

Here, both values return a new object instance and new function instance on every render, and we definitely don't want to leave/re-enter the room on every render.

So we definitely don't want to include these in the dependency array here! So what then? Do we want to disable the `exhaustive-deps` checks here in its entirety? No! Because it's still useful to keep enforcing that we don't introduce bugs here in the future, if we refactor this code. (When we add/remove dependent variables, we surely want to keep getting ESlint errors here!)

I've wanted to make this choice explicit and found a way through something I'm calling "boxing" here[^1]. By wrapping the values passed here in a mutable ref, and simply resetting that ref on every render, we get the behavior we have today but codified more explicitly, so it won't look like a bug/mistake, and we won't have to disable/ignore the `exhaustive-deps` rule here.

So effectively this behaves 100% the same as omitting the value from the deps array, like we did until now.

[^1]: Not sure if there is prior art for this concept, or if it's perhaps known by another name already?